cmd/manager/main.go

```diff
@@ -88,6 +90,27 @@ func main() {
 
 	printVersion()
 
+	// Take exclusive instance lock if required, we need that to make sure that
+	// we don't have more than one working instance of k8s-nmstate
+	if lockFilePath, ok := os.LookupEnv("NMSTATE_INSTANCE_NODE_LOCK_FILE"); ok {
+		log.Info("Try to take exclusive lock on file: %s", lockFilePath)
+		lockFileMutex, err := lockfile.New(lockFilePath)
+		if err != nil {
+			log.Error(err, "Failed to create lockFile for NMSTATE_INSTANCE_NODE_LOCK_FILE")
+			os.Exit(1)
+		}
+		for {
+			err = lockFileMutex.TryLock()
```

Comment:
We can use some syntactic sugar here:

if err = lockFileMutex.TryLock(); err == nil {
sleep
log info
}